contracts/BaseWrapper.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+
+import {RegistryAPI, VaultAPI} from "./BaseStrategy.sol";
+
+abstract contract BaseWrapper {
+    using SafeMath for uint256;
+
+    ERC20 public token;
+
+    // Reduce number of external calls (SLOADs stay the same)
+    VaultAPI[] private _cachedVaults;
+
+    // v2.registry.ychad.eth
+    RegistryAPI constant registry = RegistryAPI(0xE15461B18EE31b7379019Dc523231C57d1Cbc18c);
```

Comment:
just in case may want to make this mutable by setter instead of constant, gives us flexibility to upgrade registry as last resort, guessing is less of  pain to migrate the registry than M wrappers

Reply:
fair

Reply:
Fixed in e4e2418